Sources/Fullscreen/LoadingView/LoadingView.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import UIKit
+
+/// Branded alternative of SVProgressHUD. It's presented on top of the top-most view.
+public class LoadingView: UIView {
+    /// Allows the loading view to use a plain UIActivityIndicatorView,
+    /// useful for a smooth transition between the old indicator and the new one,
+    /// by using this flag we can avoid having multiple styles of showing progress in our app.
+    public static let shouldUseOldIndicator: Bool = false
+
+    private let animationDuration: TimeInterval = 0.3
+    private let loadingIndicatorSize: CGFloat = 40
+    private let loadingIndicatorInitialTransform: CGAffineTransform = CGAffineTransform(scaleX: 0.7, y: 0.7)
+    private static let shared = LoadingView()
+
+    private lazy var loadingIndicator: LoadingIndicatorView = {
+        let view = LoadingIndicatorView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.transform = loadingIndicatorInitialTransform
+        return view
+    }()
+
+    private lazy var oldLoadingIndicator: UIActivityIndicatorView = {
+        let view = UIActivityIndicatorView(activityIndicatorStyle: .whiteLarge)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.transform = loadingIndicatorInitialTransform
+        view.color = .primaryBlue
+        return view
+    }()
+
+    private var defaultWindow: UIWindow?
+
+    public init(window: UIWindow? = UIApplication.shared.keyWindow) {
+        super.init(frame: .zero)
+        self.defaultWindow = window
+        self.alpha = 0
+        self.translatesAutoresizingMaskIntoConstraints = false
+        setup()
+    }
+
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError()
+    }
+
+    /// Adds a layer on top of the top-most view and starts the animation of the loading indicator.
+    public class func show() {
+        LoadingView.shared.startAnimating()
+    }
+
+    /// Stops the animation of the loading indicator and removes the loading view.
+    public class func hide() {
+        LoadingView.shared.stopAnimating()
+    }
+}
+
+// MARK: - Private methods
+
+extension LoadingView {
```

Comment:
You could also make this extension private `private extension LoadingView` and remove private from methods in it.

Reply:
I made the extension private but I prefer to keep methods private unless they *need* to be internal or public.

Reply:
Yeah, but I meant that methods in `private extension` become private automatically.

Reply:
This is probably something we should add to coding guidelines then. For instance we have tried to only specify on extension level as much as possible in filters. 

But I agree, if the extension is bigger than a couple of methods it makes it harder to see quickly when it is only on extension level.